sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -197,6 +204,10 @@ String getDefaultStreamName() {
         return BigQueryHelpers.stripPartitionDecorator(tableUrn) + "/streams/_default";
       }
 
+      String getStreamAppendClientCacheEntryName() {
+        return getDefaultStreamName() + "-client" + clientNumber;
+      }
```

Comment:
This is a bit weird, since this code doesn't always use the default stream. Now the cache is probably not needed in the non default stream case (since we'll create a new stream for every bundle), however if we change that we need to rename the cache and also make sure to close the client (since right now we rely on the cache removal listener to close the client)

Reply:
I will make the changes to consider default streams and per bundle on-demand streams.